src/runtime/ggml/backend.cpp

```
@@ -44,41 +50,56 @@ BackendManager::init_backends() {
-            ggml_backend_dev_t dev_cur = ggml_backend_dev_get(i);
-            GGMLF_LOG_INFO("Device %d: %s\n", i, ggml_backend_dev_name(dev_cur));
-            const auto dev_type = ggml_backend_dev_type(dev_cur);
-            if (dev_type == GGML_BACKEND_DEVICE_TYPE_GPU ||
-                dev_type == GGML_BACKEND_DEVICE_TYPE_IGPU) {
-                // Register buffer types only for the selected device.
-                if (idx == params.gpu_device_idx) {
-                    dev = dev_cur;
-                    auto* buft = ggml_backend_dev_buffer_type(dev);
-                    if (buft) {
-                        buft_list.emplace_back(dev, buft);
+        if (borrowed_gpu_backend_ != nullptr) {
```

Clang (14.0.6)

[note] +2, including nesting penalty of 1, nesting level increased to 2

(clang)